payments-contracts/src/test/kotlin/com/gevamu/corda/test/contracts/PaymentContractSentToGatewayTest.kt

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright 2022 Exactpro Systems Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.gevamu.corda.test.contracts
+
+import com.gevamu.corda.contracts.PaymentContract
+import com.gevamu.corda.states.Payment
+import net.corda.testing.node.ledger
+import org.junit.jupiter.api.Test
+
+class PaymentContractSentToGatewayTest : AbstractPaymentContractTest() {
+    private val inputPayment = Payment(
+        uniquePaymentId = uniquePaymentId,
+        payer = payer.party,
+        gateway = gateway.party,
+        endToEndId = endToEndId,
+        paymentInstructionId = attachmentId,
+        status = Payment.PaymentStatus.CREATED
+    )
+
+    @Test
+    fun `test valid payment`() {
+        val outputPayment = Payment(
+            uniquePaymentId = uniquePaymentId,
+            payer = payer.party,
+            gateway = gateway.party,
+            endToEndId = endToEndId,
+            paymentInstructionId = attachmentId,
+            status = Payment.PaymentStatus.SENT_TO_GATEWAY
+        )
+        ledgerServices.ledger {
+            transaction {
+                command(payer.publicKey, PaymentContract.Commands.Create(uniquePaymentId))
+                output(PaymentContract.ID, Payment.PaymentStatus.CREATED.name, inputPayment)
```

Comment:
It's pretty fine to use literals in unit tests. When you write `CREATED.name` the first thought is that it has some special meaning besides of being simple label. I'd switch to string literals for state labels.